layers/dns.go

```diff
@@ -464,16 +465,32 @@ func (rr *DNSResourceRecord) String() string {
 	return "..."
 }
 
+func decodeCharacterStrings(data []byte) ([][]byte, error) {
+	var strings [][]byte
+	end := len(data)
+	for index, index2 := 0, 0; index != end; index = index2 {
+		index2 = index + 1 + int(data[index])
```

Comment:
maybe add quick comment here: Since index2 is an int{32,64} and data[index] is a uint8, there's no way we can overflow and have index2 < start.